src/openvic-simulation/Modifier.cpp

```diff
@@ -117,47 +121,105 @@ bool ModifierManager::add_modifier(std::string_view identifier, ModifierValue&&
 bool ModifierManager::setup_modifier_effects() {
 	bool ret = true;
 
+	using enum ModifierEffect::format_t;
+
 	ret &= add_modifier_effect("movement_cost", false);
 	ret &= add_modifier_effect("farm_rgo_size", true);
 	ret &= add_modifier_effect("farm_rgo_eff", true);
 	ret &= add_modifier_effect("mine_rgo_size", true);
 	ret &= add_modifier_effect("mine_rgo_eff", true);
-	ret &= add_modifier_effect("min_build_railroad", false);
-	ret &= add_modifier_effect("supply_limit", true);
+	ret &= add_modifier_effect("supply_limit", true, RAW_DECIMAL);
 	ret &= add_modifier_effect("combat_width", false);
-	ret &= add_modifier_effect("defence", true);
+	ret &= add_modifier_effect("defence", true, RAW_DECIMAL);
+	ret &= add_modifier_effect("local_ship_build", false);
+	ret &= add_modifier_effect("research_points_modifier", true);
+	ret &= add_modifier_effect("local_rgo_output", true);
+	ret &= add_modifier_effect("attrition", false, RAW_DECIMAL);
+	ret &= add_modifier_effect("immigrant_push", false);
+	ret &= add_modifier_effect("population_growth", true);
+	ret &= add_modifier_effect("local_RGO_throughput", true);
+	ret &= add_modifier_effect("assimilation_rate", true);
```

Comment:
do you guys need a list of what the possible modifiers are? (there are two sets, one for province modifiers and one for national modifiers)

Reply:
Sure, we’ve got lists based off of the wiki but I can expect that’s far from perfect.

Reply:
Some items in this list are, as noted, alternate spellings for other items. Items marked as (p) affect provinces. Items marked as (n) affect nations. If an items marked (n) is part of a modifier given to a province it will instead affect the nation owning the province. An item marked (b) can serve as both: if it is in a modifier applied to a province, it affects the province, otherwise it affects the nation.

```
	supply_limit (p)
	attrition (p)
	max_attrition (p)
	war_exhaustion (n)
	max_war_exhaustion (n)
	leadership (n)
	leadership_modifier (n)
	supply_consumption (n)
	org_regain (n)
	reinforce_speed (n)
	land_organisation (n)
	land_organization (alt for land_organisation)
	army_organisation (alt for land_organisation)
	army_organization (alt for land_organisation)
	naval_organisation (n)
	naval_organization (alt for naval_organisation)
	research_points (n)
	research_points_modifier (n)
	research_points_on_conquer (n)
	import_cost (n)
	loan_interest (n)
	tax_efficiency (n)
	min_tax (n)
	max_tax (n)
	min_military_spending (n)
	max_military_spending (n)
	min_social_spending (n)
	max_social_spending (n)
	factory_owner_cost (n)
	local_factory_owner_cost (p)
	min_tariff (n)
	max_tariff (n)
	ruling_party_support (n)
	local_ruling_party_support (p)
	rich_vote (n)
	middle_vote (n)
	poor_vote (n)
	minimum_wage (n)
	factory_maintenance (n)
	poor_life_needs (b)
	rich_life_needs (b)
	middle_life_needs (b)
	poor_everyday_needs (b)
	rich_everyday_needs (b)
	middle_everyday_needs (b)
	poor_luxury_needs (b)
	middle_luxury_needs (b)
	rich_luxury_needs (b)
	unemployment_benefit (n)
	pension_level (n)
	population_growth (p)
	global_population_growth (n)
	factory_input (n)
	factory_output (n)
	factory_throughput (n)
	rgo_input (n)
	rgo_output (n)
	rgo_throughput (n)
	artisan_input (n)
	artisan_output (n)
	artisan_throughput (n)
	local_factory_input (p)
	local_factory_output (p)
	local_factory_throughput (p)
	local_rgo_input (p)
	local_rgo_output (p)
	local_rgo_throughput (p)
	local_artisan_input (p)
	local_artisan_output (p)
	local_artisan_throughput (p)
	number_of_voters (p)
	goods_demand (b)
	badboy (n)
	assimilation_rate (b)
	global_assimilation_rate (n)
	prestige (n)
	factory_cost (n)
	life_rating (p)
	farm_rgo_eff (b)
	mine_rgo_eff (b)
	farm_rgo_size (b)
	mine_rgo_size (b)
	rgo_size (b)
	issue_change_speed (n)
	social_reform_desire (n)
	political_reform_desire (n)
	literacy_con_impact (n)
	pop_militancy_modifier (b)
	pop_consciousness_modifier (b)
	rich_income_modifier (b)
	middle_income_modifier (b)
	poor_income_modifier (b)
	low_income_modifier (alt for poor_income_modifier)
	boost_strongest_party (p)
	global_immigrant_attract (n)
	immigration (alt for immigrant_attract)
	immigrant_attract (p)
	immigrant_push (p)
	local_repair (p)
	local_ship_build (p)
	poor_savings_modifier (n)
	middle_savings_modifier (n)
	rich_savings_modifier (n)
	influence_modifier (n)
	diplomatic_influence (alt for influence_modifier)
	diplomatic_points_modifier (n)
	mobilisation_size (alt for mobilization_size)
	mobilisation_economy_impact (alt for mobilization_economy_impact)
	mobilization_size (n)
	mobilization_economy_impact (n)
	global_pop_militancy_modifier (n)
	global_pop_consciousness_modifier (n)
	core_pop_militancy_modifier (n)
	core_pop_consciousness_modifier (n)
	non_accepted_pop_militancy_modifier (n)
	non_accepted_pop_consciousness_modifier (n)
	cb_generation_speed_modifier (n)
	mobilization_impact (n)
	mobilisation_impact (alt for mobilization_impact)
	suppression_points_modifier (n)
	education_efficiency_modifier (n)
	education_efficiency (n)
	education (alt for education_efficiency_modifier)
	civilization_progress_modifier (n)
	administrative_efficiency_modifier (n)
	land_unit_start_experience (n)
	naval_unit_start_experience (n)
	naval_attack_modifier (n)
	naval_defense_modifier (n)
	naval_defence_modifier (alt for naval_defense_modifier)
	land_attack_modifier (n)
	land_defense_modifier (n)
	land_defence_modifier (alt for land_defense_modifier)
	tariff_efficiency_modifier (n)
	max_loan_modifier (n)
	movement_cost (p)
	unciv_economic_modifier (n)
	unciv_military_modifier (n)
	self_unciv_economic_modifier (n)
	self_unciv_military_modifier (n)
	defender (alt for defence)
	attacker (alt for attack)
	defence (p)
	defense (alt for defence)
	attack (p)
	combat_width (p)
	min_build_naval_base (p)
	min_build_railroad (p)
	min_build_fort (p)
	min_build_bank (p)
	min_build_university (p)
	commerce_tech_research_bonus (n)
	army_tech_research_bonus (n)
	industry_tech_research_bonus (n)
	navy_tech_research_bonus (n)
	culture_tech_research_bonus (n)
	colonial_migration (n)
	max_national_focus (n)
	cb_creation_speed (alt for cb_generation_speed_modifier)
	diplomatic_points (alt for diplomatic_points_modifier)
	reinforce_rate (n)
	tax_eff (alt for tax_efficiency)
	administrative_efficiency (alt for administrative_efficiency_modifier)
	admin_efficiency (alt for administrative_efficiency_modifier)
	influence (n)
	dig_in_cap (n)
	morale (alt for org_regain)
	military_tactics (n)
	supply_range (n)
	regular_experience_level (n)
	increase_research (alt for research_points_modifier)
	permanent_prestige (n)
	soldier_to_pop_loss (n)
	naval_attrition (n)
	land_attrition (n)
	pop_growth (n)
	global_pop_growth (alt for pop_growth)
	colonial_life_rating (n)
	seperatism (n)
	colonial_prestige (n)
	unit_start_experience (n)
	unit_recruitment_time  (n)
```